src/getRepositoryName.ts

```diff
@@ -16,12 +16,17 @@ import { PrismicError } from "./errors/PrismicError"
  */
 export const getRepositoryName = (repositoryEndpoint: string): string => {
 	try {
-		return new URL(repositoryEndpoint).hostname.split(".")[0]
-	} catch {
-		throw new PrismicError(
-			`An invalid Prismic Rest API V2 endpoint was provided: ${repositoryEndpoint}`,
-			undefined,
-			undefined,
-		)
-	}
+		const parts = new URL(repositoryEndpoint).hostname.split(".")
+
+		// [subdomain, domain, tld]
+		if (parts.length > 2) {
+			return parts[0]
+		}
```

Comment:
It's to prevent attempting to infer the repository domain from proxied endpoints when they don't have a subdomain `example.com` would infer `example` as the repository name.

I agree this is odd, I encountered it there: https://github.com/prismicio/prismic-client/commit/96bce5f4c37d622732c8c357b298a438ce2d30c6#diff-d02780c2f062abcf8505db670090d74c454de33dcf58974ba92e68db603208a2 Technically we're the only one ever needed that when testing on staging...

Reply:
Reword to check it's a Prismic domain (prod, stage, dev)